src/unstract/adapters/vectordb/pinecone/src/pinecone.py

```diff
@@ -68,6 +75,22 @@ def get_vector_db_instance(self) -> BasePydanticVectorStore:
                 VectorDbConstants.EMBEDDING_DIMENSION,
                 VectorDbConstants.DEFAULT_EMBEDDING_SIZE,
             )
+
+            specification = self.config.get(Constants.SPECIFICATION)
+            if specification == Constants.SPEC_POD:
+                environment = self.config.get(Constants.ENVIRONMENT)
+                spec = PodSpec(
+                    environment=environment,
+                    replicas=Constants.DEFAULT_SPEC_COUNT_VALUE,
```

Comment:
@gaya3-zipstack we can leverage JSON schema's validation to apply the default values our schemas specify. We have handled this for [tool settings in the SDK](https://github.com/Zipstack/unstract-sdk/blob/c46fdc075e5e752b659599b6d92b5e0367265c1b/src/unstract/sdk/tool/validator.py#L92-L111). We could leverage the same [DefaultsGeneratingValidator](https://github.com/Zipstack/unstract-sdk/blob/c46fdc075e5e752b659599b6d92b5e0367265c1b/src/unstract/sdk/tool/validator.py#L49) at a generic place so that it applies to all adapters - maybe at the backend/adapter base class constructor.

This can help avoid duplicating logic/efforts in both JSON schema and the code. This might be a bigger change - feel free to leverage this based on need and effort to implement